source/module.c

```diff
@@ -125,6 +125,11 @@ uint32_t PyObject_GetAttrAsUint32(PyObject *o, const char *class_name, const cha
         return result;
     }
 
+    if (attr == Py_None) {
+        PyErr_Format(PyExc_AttributeError, "'%s.%s' required integral attribute is None", class_name, attr_name);
+        return result;
```

Comment:
attr is a [new reference](https://docs.python.org/3/c-api/object.html#c.PyObject_GetAttrString)

need to decref it. maybe:

```suggestion
            goto done;
```
and then later
```
done:
    Py_DECREF(attr);
    return result;
}
```

that's how the [old versions](https://github.com/awslabs/aws-crt-python/blob/b27e0a70973c2f62feec123886dc30ac027b296c/source/module.c#L145) of these functions used to do it

Reply:
If you really want to be a champ. Edit 
```
PyObject *attr = PyObject_GetAttrString(o, attr_name);
```
in all these functions to be like:
```
PyObject *attr = PyObject_GetAttrString(o, attr_name); /* new reference */
```

I mean, that's what we advise in our [coding guidelines](https://github.com/awslabs/aws-crt-python/blob/main/guides/dev/writing-code.md#reference-counting-in-c) (which were written years after most of this C code)

Reply:
But it's Py_None in the case you're worried about.

I could add a comment saying why there's no need to decref it.

Reply:
But PyObject_GetAttrString() still increments the refcount on it. Seems weird to leak a reference on purpose because you think it probably doesn't matter. Vs just doing the normal refcount thing